govcd/vm_test.go

```diff
@@ -2281,3 +2281,55 @@ func createNsxtVAppAndVm(vcd *TestVCD, check *C) (*VApp, *VM) {
 
 	return vapp, vm
 }
+
+func (vcd *TestVCD) Test_GetOvfEnvironment(check *C) {
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vapp was not successfully created at setup")
+	}
+	vapp := vcd.findFirstVapp()
+	existingVm, vmName := vcd.findFirstVm(vapp)
+	if vmName == "" {
+		check.Skip("skipping test because no VM is found")
+	}
+	vm, err := vcd.client.Client.GetVMByHref(existingVm.HREF)
+	check.Assert(err, IsNil)
+
+	vmStatus, err := vm.GetStatus()
+	check.Assert(err, IsNil)
+	if vmStatus != "POWERED_ON" {
+		task, err := vm.PowerOn()
+		check.Assert(err, IsNil)
+		err = task.WaitTaskCompletion()
+		check.Assert(err, IsNil)
+		check.Assert(task.Task.Status, Equals, "success")
+	}
+
+	// Read ovfenv when VM is started
+	ovfenv, err := vm.GetEnvironment()
+	check.Assert(err, IsNil)
+	check.Assert(strings.Contains(ovfenv.VCenterId, "vm-"), Equals, true)
+	check.Assert(ovfenv, NotNil)
+	check.Assert(ovfenv.PlatformSection, NotNil)
+	check.Assert(ovfenv.PlatformSection.Kind, Equals, "VMware ESXi")
```

Comment:
You are assuming that the VM is from VMware. What happens if it was a plain Ubuntu or CentOS?

Reply:
PlatformSection show hypervisor informations.

So don't worry about Guest VM opertaing system.

Reply:
https://www.dmtf.org/standards/ovf

https://www.dmtf.org/sites/default/files/standards/documents/DSP0243_2.1.1.pdf
page 40


The **PlatformSection** element contains optional information provided by the deployment function. The
Kind, Version, and Vendor elements **describe the virtualization platform**. The Locale and TimeZone
elements describe the current locale and time zone.